### PR TITLE
Removing notification settings v1

### PR DESCRIPTION
**Pull request by ouchadam**
Removes the notification settings v1 code branches and flag

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/home/room/list/actions/RoomListQuickActionsEpoxyController.kt [hunk 1]
```diff
@@ -48,54 +46,38 @@ class RoomListQuickActionsEpoxyController @Inject constructor(
...
-                stringProvider(host.stringProvider)
-                colorProvider(host.colorProvider)
-                izLowPriority(roomSummary.isLowPriority)
-                izFavorite(roomSummary.isFavorite)
-                settingsClickListener { host.listener?.didSelectMenuAction(RoomListQuickActionsSharedAction.Settings(roomSummary.roomId)) }
-                favoriteClickListener { host.listener?.didSelectMenuAction(RoomListQuickActionsSharedAction.Favorite(roomSummary.roomId)) }
-                lowPriorityClickListener { host.listener?.didSelectMenuAction(RoomListQuickActionsSharedAction.LowPriority(roomSummary.roomId)) }
-            }
+        // Preview, favorite, settings
+        bottomSheetRoomPreviewItem {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/list/actions/RoomListQuickActionsEpoxyController.kt:50]: Test `state.roomListActionsArgs.mode == RoomListActionsArgs.Mode.FULL` should stay

File: vector/src/main/java/im/vector/app/features/home/room/list/actions/RoomListQuickActionsEpoxyController.kt [hunk 2]
```diff
@@ -48,54 +46,38 @@ class RoomListQuickActionsEpoxyController @Inject constructor(
...
-            RoomListQuickActionsSharedAction.NotificationsAllNoisy(roomSummary.roomId).toBottomSheetItem(0, selectedRoomState)
-            RoomListQuickActionsSharedAction.NotificationsAll(roomSummary.roomId).toBottomSheetItem(1, selectedRoomState)
-            RoomListQuickActionsSharedAction.NotificationsMentionsOnly(roomSummary.roomId).toBottomSheetItem(2, selectedRoomState)
-            RoomListQuickActionsSharedAction.NotificationsMute(roomSummary.roomId).toBottomSheetItem(3, selectedRoomState)
         }
 
-        if (showFull) {
-            RoomListQuickActionsSharedAction.Leave(roomSummary.roomId, showIcon = !isV2).toBottomSheetItem(5)
-        }
+        RoomListQuickActionsSharedAction.Leave(roomSummary.roomId, showIcon = !true).toBottomSheetItem(5)
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/list/actions/RoomListQuickActionsEpoxyController.kt:80]: So please also revert this change

**Review by bmarty — CHANGES_REQUESTED** [review]
You have done too much cleanup, see my other comment :)

File: vector/src/main/java/im/vector/app/features/home/room/list/actions/RoomListQuickActionsEpoxyController.kt (hunk 1 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/home/room/list/actions/RoomListQuickActionsEpoxyController.kt:50]: Ah maybe not, it seems that only FULL is used nowadays.
Maybe cleanup https://github.com/vector-im/element-android/blob/main/vector/src/main/java/im/vector/app/features/home/room/list/actions/RoomListQuickActionsBottomSheet.kt#L50

Comment (ouchadam) [reply vector/src/main/java/im/vector/app/features/home/room/list/actions/RoomListQuickActionsEpoxyController.kt:50]: the previous condition was 

```kotlin
// V2 always shows full details as we no longer display the sheet from RoomProfile > Notifications
val showFull = state.roomListActionsArgs.mode == RoomListActionsArgs.Mode.FULL || isV2
```` 

now that v2 is always true, do we need the condition (perhaps it was wrong previously :thinking: )

Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/home/room/list/actions/RoomListQuickActionsEpoxyController.kt:50]: I do not think it was wrong before, it was OK in v1 mode, but [with this PR](https://github.com/vector-im/element-android/pull/4627/files#diff-0ed745b1f18ce801155cc8f6f7ff155ce1741a3c63e927a86fee0d46583bb425L265), `RoomListActionsArgs.Mode.NOTIFICATIONS` is never used, so maybe just delete `RoomListActionsArgs.Mode`.

File: vector/src/main/java/im/vector/app/features/home/room/list/actions/RoomListQuickActionsEpoxyController.kt (hunk 2 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/home/room/list/actions/RoomListQuickActionsEpoxyController.kt:80]: (ignore)

File: vector/src/main/java/im/vector/app/features/home/room/list/actions/RoomListQuickActionsEpoxyController.kt (hunk 1 above)
Comment (ouchadam) [reply vector/src/main/java/im/vector/app/features/home/room/list/actions/RoomListQuickActionsEpoxyController.kt:50]: will do!

Comment (ouchadam) [reply vector/src/main/java/im/vector/app/features/home/room/list/actions/RoomListQuickActionsEpoxyController.kt:50]: removed c208c2d5ae9e6c6158756b654588eee096a57996

File: vector/src/main/java/im/vector/app/features/home/room/list/actions/RoomListQuickActionsEpoxyController.kt (hunk 2 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/home/room/list/actions/RoomListQuickActionsEpoxyController.kt:80]: I think more cleanup can be done in the fun `RoomListQuickActionsSharedAction.toBottomSheetItem(...)`, there is some dead code now.

Comment (ouchadam) [reply vector/src/main/java/im/vector/app/features/home/room/list/actions/RoomListQuickActionsEpoxyController.kt]: good catch! removed https://github.com/vector-im/element-android/pull/4627/commits/ab04e4078460c462d7ccd8f41c2cb479e524644b

**Review by bmarty — APPROVED** [review]
Thanks for the update!
